draft-schwartz-ds-glue.md

```diff
@@ -255,7 +260,7 @@ Nameservers supporting authenticated encryption MAY indicate any DANE mode, or n
 
 As an optimization, nameservers using DANE MAY place a TLSA record in the DSGLUE to avoid the latency of a TLSA lookup during delegation.  However, child zones should be aware that this adds complexity and delay to the process of TLSA key rotation.
```

Comment:
I would prefer to not allow adding TLSA RRs to the DSGLUE (other than the presence/absence bit).

Reply:
Interesting.  Why do you think it should be disallowed?  The text doesn't recommend  it, but it seems harmless (and a valuable optimization) for zones that are willing to deal with the maintenance overhead (which is really quite minimal when using CDS).

Recipients (i.e. resolvers) are free to ignore nonempty TLSA DSGLUE.  If it helps, we could note that in the text.

Reply:
> Interesting. Why do you think it should be disallowed? The text doesn't recommend it, but it seems harmless (and a valuable optimization) for zones that are willing to deal with the maintenance overhead (which is really quite minimal when using CDS).
> 
> Recipients (i.e. resolvers) are free to ignore nonempty TLSA DSGLUE. If it helps, we could note that in the text.

On the principle that adding features is easier than removing features. It is not clear yet that TLSA is essential and can be kept up to date by child zone operators. Should be added if there is consensus on the need for it.

Reply:
I've added a note here discussing the open question about what to recommend:

> QUESTION: Should we recommend for or against including nonempty TLSA in DSGLUE?  If CDS-like update mechanisms work well, and ADoT-DANE is widely deployed, this could warrant a positive recommendation.  Conversely, if rotation is error-prone, and ADoT-DANE is rare, a negative recommendation might be better.

I actually think "removing" this option (i.e. deprecating it and ignoring any TLSA DSGLUE) would likely be easier than adding it in the future, due to the potential for systems rejecting unexpected new RR types.

Given the wide range of opinions about DANE generally, I don't expect we'll ever see a clear consensus about what is needed, but hopefully we can get more input from the working group.